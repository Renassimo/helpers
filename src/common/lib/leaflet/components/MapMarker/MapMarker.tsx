import { Marker, Popup } from 'react-leaflet';
import { ReactNode } from 'react';

import getMarkerIcon from '../../utils/getMarkerIcon';

const MapMarker = ({
  color,
  position,
  isNew = false,
  children,
}: {
  color: string;
  position: [number, number];
  isNew?: boolean;
  children?: ReactNode;
}) => {
  return (
    <>
      <Marker position={position} icon={getMarkerIcon(color, isNew)}>
        {children && <Popup>{children}</Popup>}
      </Marker>
    </>
  );
};

export default MapMarker;
