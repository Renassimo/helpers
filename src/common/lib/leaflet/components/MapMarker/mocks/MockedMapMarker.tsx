const MockedMapMarker = jest.fn(
  ({ color, position, isNew, isMarked, children }) => {
    return (
      <div>
        {color} - {position?.[0]} - {position?.[1]} -{' '}
        {isNew ? 'new' : 'existing'} - {isMarked ? 'marked' : 'not marked'} -{' '}
        {children}
      </div>
    );
  }
);

export default MockedMapMarker;
