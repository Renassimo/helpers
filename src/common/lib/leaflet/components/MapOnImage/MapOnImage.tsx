import { MapContainer, ImageOverlay, ZoomControl } from 'react-leaflet';

import Box from '@mui/material/Box';

import MapEventsHandler from '@/common/lib/leaflet/components/MapEventsHandler';
import getMapBoundsAndCenter from '@/common/lib/leaflet/utils/getMapBoundsAndCenter';

import { ReactNode } from 'react';
import { LeafletMouseEvent } from 'leaflet';

import 'leaflet/dist/leaflet.css';

const MapOnImage = ({
  onClick,
  mapImageUrl,
  backgroundColor,
  mapImageRatio,
  children,
}: {
  onClick: (event: LeafletMouseEvent) => void;
  mapImageUrl?: string;
  backgroundColor?: string;
  mapImageRatio?: number | null;
  children: ReactNode;
}) => {
  const yRange = mapImageRatio ? 1 / mapImageRatio : 1;
  const { center, bounds } = getMapBoundsAndCenter({ yRange });

  return (
    <Box width="100%" height="100%">
      {mapImageUrl && (
        <MapContainer
          center={center}
          zoom={10}
          style={{
            height: '100%',
            width: '100%',
            backgroundColor,
          }}
          attributionControl={false}
          zoomControl={false}
        >
          <ImageOverlay
            url={mapImageUrl}
            bounds={bounds}
            opacity={1}
            zIndex={10}
          />
          <ZoomControl position="bottomleft" />
          <MapEventsHandler handleMapClick={onClick} />
          {children}
        </MapContainer>
      )}
    </Box>
  );
};

export default MapOnImage;
