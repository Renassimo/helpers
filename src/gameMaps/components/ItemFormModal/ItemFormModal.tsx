import { useEffect } from 'react';

import useItemForm from '@/gameMaps/hooks/useItemForm';

import Modal from '@/common/components/Modal';
import ItemForm from '@/gameMaps/components/ItemForm';

import { ItemData } from '@/gameMaps/types';

const ItemFormModal = ({
  isModalOpen,
  setIsModalOpen,
  data,
  onFinish,
  gameId,
  coordinates,
  playId,
  categoryId,
  clearData,
}: {
  isModalOpen: boolean;
  setIsModalOpen: (isOpen: boolean) => void;
  data?: ItemData | null;
  onFinish: (newData: ItemData | null, id?: string) => void;
  gameId: string;
  coordinates: [number, number];
  playId: string;
  categoryId: string;
  clearData?: () => void;
}) => {
  const onModalClose = () => {
    setIsModalOpen(false);
    cleanForm();
  };

  const handleFinish = (newData: ItemData | null) => {
    onFinish(newData, data?.id);
    onModalClose();
  };

  const {
    isEditForm,
    values,
    setters,
    cleanForm,
    prepareFormForEdit,
    onSubmit,
    onDelete,
    errors,
    loading,
  } = useItemForm(
    { gameId, playId, categoryId, coordinates },
    data,
    handleFinish
  );

  const handleDelete = async (): Promise<void> => {
    if (confirm(`Are you sure to delete this item?`)) {
      await onDelete();
      handleFinish(null);
    }
  };

  useEffect(() => {
    if (isModalOpen && isEditForm) {
      prepareFormForEdit();
    }
  }, [isModalOpen, isEditForm]);

  // Clears data when closes modal
  useEffect(() => {
    if (!isModalOpen && isEditForm) {
      clearData?.();
    }
  }, [isModalOpen, isEditForm]);

  return (
    <Modal
      open={isModalOpen}
      onClose={onModalClose}
      onSubmit={onSubmit}
      title={`${isEditForm ? 'Update' : `Create new`} item`}
      loading={loading}
    >
      <ItemForm
        values={values}
        setters={setters}
        errors={errors}
        onDelete={isEditForm ? handleDelete : undefined}
        isReady={isModalOpen}
      />
    </Modal>
  );
};

export default ItemFormModal;
