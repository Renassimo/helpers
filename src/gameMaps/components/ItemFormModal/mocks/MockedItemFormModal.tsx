const MockedItemFormModal = jest.fn(
  ({ isModalOpen, gameId, coordinates, playId, categoryId }) => (
    <div>
      MockedItemFormModal {isModalOpen ? 'opened' : 'closed'} - {gameId} -{' '}
      {coordinates?.[0]} - {coordinates?.[1]} - {playId} - {categoryId}
    </div>
  )
);

export default MockedItemFormModal;
