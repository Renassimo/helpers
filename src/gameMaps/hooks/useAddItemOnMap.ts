import { useCallback, useEffect, useMemo, useState } from 'react';

import useAlerts from '@/common/hooks/alerts';

import renderSaveMarkerPopupContent from '@/common/lib/leaflet/utils/renderSaveMarkerPopupContent';

import { CategoriesState, ItemData, ItemMarker } from '@/gameMaps/types';
import { LeafletMouseEvent } from 'leaflet';

const useAddItemOnMap = ({
  categories,
  pointingCategoryId,
  visibleItems,
  onAdd,
}: {
  categories: CategoriesState;
  pointingCategoryId: string | null;
  visibleItems: ItemData[];
  onAdd: (coordinates: [number, number]) => void;
}): {
  allMarkers: (ItemData | ItemMarker | null)[];
  handleMapClick: (event: LeafletMouseEvent) => void;
  newMarker: ItemMarker | null;
} => {
  const [newMarker, setNewMarker] = useState<ItemMarker | null>(null);
  const allMarkers = useMemo(
    () => [...(visibleItems ?? []), newMarker],
    [visibleItems, newMarker]
  );

  const { createInfoAlert, clearAll } = useAlerts();

  const deleteNewMarker = useCallback(() => {
    clearAll();
    setNewMarker(null);
  }, []);

  const openCreateNewItem = useCallback((coordinates: [number, number]) => {
    clearAll();
    onAdd(coordinates);
  }, []);

  const renderCreateItemPopupContent = useCallback(
    (coordinates: [number, number]) =>
      renderSaveMarkerPopupContent({
        text: `Add new ${
          pointingCategoryId
            ? categories[pointingCategoryId].attributes.title
            : ''
        } item?`,
        onAdd: () => openCreateNewItem(coordinates),
        onCancel: deleteNewMarker,
      }),
    [categories, pointingCategoryId]
  );

  const handleMapClick = useCallback(
    (event: LeafletMouseEvent) => {
      if (!pointingCategoryId) return;

      const { lat, lng } = event.latlng;
      const coordinates: [number, number] = [lat, lng];

      setNewMarker({
        attributes: {
          coordinates,
          description: renderCreateItemPopupContent(coordinates),
          categoryId: pointingCategoryId,
        },
      });
      clearAll();
      createInfoAlert(renderCreateItemPopupContent(coordinates), 0);
    },
    [pointingCategoryId]
  );

  useEffect(() => {
    if (pointingCategoryId) {
      clearAll();
      createInfoAlert(
        `Point on map to create a new ${categories[pointingCategoryId].attributes.title} item`,
        0
      );
    }
  }, [categories, pointingCategoryId]);

  useEffect(() => {
    if (!pointingCategoryId) deleteNewMarker();
  }, [pointingCategoryId]);

  return { allMarkers, handleMapClick, newMarker };
};

export default useAddItemOnMap;
