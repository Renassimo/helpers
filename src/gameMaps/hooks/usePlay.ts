import { useContext } from 'react';

import PlayContext from '@/gameMaps/contexts/PlayContext';

const usePlay = () => useContext(PlayContext);

export default usePlay;
