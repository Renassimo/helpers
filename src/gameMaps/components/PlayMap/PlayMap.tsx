import { MouseEvent } from 'react';

import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import EditIcon from '@mui/icons-material/Edit';
import EditLocationAltIcon from '@mui/icons-material/EditLocationAlt';

import MapOnImage from '@/common/lib/leaflet/components/MapOnImage';
import MapMarker from '@/common/lib/leaflet/components/MapMarker';
import ItemFormModal from '@/gameMaps/components/ItemFormModal';

import usePlay from '@/gameMaps/hooks/usePlay';
import useAddItemOnMap from '@/gameMaps/hooks/useAddItemOnMap';

const PlayMap = () => {
  const {
    game,
    play,
    visibleItems,
    items,
    categories,
    pointingCategoryId,
    openItemCreating,
    openItemUpdating,
    isItemEditOpen,
    setIsItemEditOpen,
    updateSubmittedItem,
    editingItem,
    relocateItem,
    relocatingItem,
    updateItemCoordinates,
  } = usePlay();
  const { handleMapClick, newMarker, relocatingMarker } = useAddItemOnMap({
    categories,
    pointingCategoryId,
    onAdd: openItemCreating,
    relocatingItem,
    updateItemCoordinates,
  });

  const { mapImageUrl, backgroundColor, mapImageRatio } =
    game?.attributes ?? {};

  const handleEditItem = (event: MouseEvent, id?: string) => {
    event.stopPropagation();
    if (id) openItemUpdating(id);
  };

  const handleRelocateItem = (event: MouseEvent, id?: string) => {
    event.stopPropagation();
    if (id) relocateItem(id);
  };

  const gameId = game?.id;
  const categoryId =
    pointingCategoryId ||
    (editingItem && items[editingItem?.id].attributes.categoryId);
  const playId = play?.id;

  const { coordinates: newMarkerCoordinates } = newMarker?.attributes ?? {};
  const { coordinates: editingItemCoordinates } = editingItem?.attributes ?? {};
  const coordinates = newMarkerCoordinates || editingItemCoordinates;

  const isItemModalAllowed = !!(gameId && categoryId && playId && coordinates);

  return (
    <>
      <MapOnImage
        onClick={handleMapClick}
        mapImageUrl={mapImageUrl}
        mapImageRatio={mapImageRatio}
        backgroundColor={backgroundColor}
      >
        {visibleItems.map((item) => (
          <MapMarker
            key={item.id}
            position={item.attributes.coordinates}
            color={categories[item.attributes.categoryId]?.attributes.color}
            isMarked={!!item.attributes.collected}
          >
            {
              <Box>
                <Box>
                  <Typography>{item.attributes.description}</Typography>
                </Box>
                <Box textAlign="right">
                  <IconButton
                    aria-label="edit-coordinates"
                    size="small"
                    onClick={(event) => handleRelocateItem(event, item.id)}
                  >
                    <EditLocationAltIcon />
                  </IconButton>
                  <IconButton
                    aria-label="edit-item"
                    size="small"
                    onClick={(event) => handleEditItem(event, item.id)}
                  >
                    <EditIcon />
                  </IconButton>
                </Box>
              </Box>
            }
          </MapMarker>
        ))}
        {newMarker && (
          <MapMarker
            key={newMarker.id}
            position={newMarker.attributes.coordinates}
            color={
              categories[newMarker.attributes.categoryId]?.attributes.color
            }
            isNew
          >
            {newMarker.attributes.description}
          </MapMarker>
        )}
        {relocatingMarker && (
          <MapMarker
            key={relocatingMarker.id}
            position={relocatingMarker.attributes.coordinates}
            color={
              categories[relocatingMarker.attributes.categoryId]?.attributes
                .color
            }
            isNew
          >
            {relocatingMarker.attributes.description}
          </MapMarker>
        )}
      </MapOnImage>
      {isItemModalAllowed && (
        <ItemFormModal
          isModalOpen={isItemEditOpen}
          setIsModalOpen={setIsItemEditOpen}
          onFinish={updateSubmittedItem}
          gameId={gameId}
          categoryId={categoryId}
          playId={playId}
          coordinates={coordinates}
          data={editingItem}
        />
      )}
    </>
  );
};

export default PlayMap;
