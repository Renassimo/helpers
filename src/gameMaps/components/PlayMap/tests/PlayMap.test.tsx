import userEvent from '@testing-library/user-event';
import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import MapOnImage from '@/common/lib/leaflet/components/MapOnImage';
import MapMarker from '@/common/lib/leaflet/components/MapMarker';

import usePlay from '@/gameMaps/hooks/usePlay';
import useAddItemOnMap from '@/gameMaps/hooks/useAddItemOnMap';
import ItemFormModal from '@/gameMaps/components/ItemFormModal';

import MockedMapOnImage from '@/common/lib/leaflet/components/MapOnImage/mocks';
import MockedMapMarker from '@/common/lib/leaflet/components/MapMarker/mocks';
import MockedItemFormModal from '@/gameMaps/components/ItemFormModal/mocks';
import {
  mockedCategory1,
  mockedCategory2,
  mockedGame,
  mockedItem1,
  mockedItem2,
  mockedItems,
  mockedPlay,
} from '@/gameMaps/types/mocks';

import PlayMap from '../PlayMap';

jest.mock('@/common/lib/leaflet/components/MapOnImage');
jest.mock('@/common/lib/leaflet/components/MapMarker');
jest.mock('@/gameMaps/components/ItemFormModal');
jest.mock('@/gameMaps/hooks/usePlay');
jest.mock('@/gameMaps/hooks/useAddItemOnMap');

describe('PlayMap', () => {
  const mockedPointingCategoryId = mockedCategory2.id;
  const mockedVisibleItems = mockedItems;
  const itemsState = {
    [mockedItem1.id]: mockedItem1,
    [mockedItem2.id]: mockedItem2,
  };
  const mockedCategoriesState = {
    [mockedCategory1.id]: mockedCategory1,
    [mockedCategory2.id]: mockedCategory2,
  };
  const mockedOpenItemCreating = jest.fn();
  const mockedOpenItemUpdating = jest.fn();
  const mockedSetIsItemEditOpen = jest.fn();
  const mockedUpdateSubmittedItem = jest.fn();
  const mockedUsePlayResult = {
    game: mockedGame,
    play: mockedPlay,
    visibleItems: mockedVisibleItems,
    items: itemsState,
    categories: mockedCategoriesState,
    pointingCategoryId: null,
    openItemCreating: mockedOpenItemCreating,
    openItemUpdating: mockedOpenItemUpdating,
    isItemEditOpen: false,
    setIsItemEditOpen: mockedSetIsItemEditOpen,
    updateSubmittedItem: mockedUpdateSubmittedItem,
    editingItem: null,
  };
  const mockedUsePlay = jest.fn(() => mockedUsePlayResult);
  const mockedHandleMapClick = jest.fn();
  const mockedNewMarker = {
    attributes: {
      coordinates: [1, 2],
      description: 'new marker description',
      categoryId: mockedPointingCategoryId,
    },
  };
  const mockedAllMarkers = mockedItems;
  const mockedUseAddItemOnMapResult = {
    allMarkers: mockedAllMarkers,
    handleMapClick: mockedHandleMapClick,
    newMarker: null,
  };
  const mockedUseAddItemOnMap = jest.fn(() => mockedUseAddItemOnMapResult);

  beforeEach(() => {
    (MapOnImage as unknown as jest.Mock).mockImplementation(MockedMapOnImage);
    (MapMarker as unknown as jest.Mock).mockImplementation(MockedMapMarker);
    (ItemFormModal as unknown as jest.Mock).mockImplementation(
      MockedItemFormModal
    );
    (usePlay as unknown as jest.Mock).mockImplementation(mockedUsePlay);
    (useAddItemOnMap as unknown as jest.Mock).mockImplementation(
      mockedUseAddItemOnMap
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(<PlayMap />);
    // Assert
    expect(baseElement).toMatchSnapshot();
    expect(mockedUsePlay).toHaveBeenCalledWith();
    expect(mockedUseAddItemOnMap).toHaveBeenCalledWith({
      categories: mockedCategoriesState,
      pointingCategoryId: null,
      visibleItems: mockedVisibleItems,
      onAdd: expect.any(Function),
    });
  });

  describe('when open for editing', () => {
    test('renders successfully with mocked modal', () => {
      // Arange
      const mockedUsePlay = jest.fn(() => ({
        ...mockedUsePlayResult,
        editingItem: mockedItem1,
      }));
      (usePlay as unknown as jest.Mock).mockImplementation(mockedUsePlay);
      // Act
      const { baseElement } = renderWithTheme(<PlayMap />);
      // Assert
      expect(baseElement).toMatchSnapshot();
      expect(mockedUsePlay).toHaveBeenCalledWith();
      expect(mockedUseAddItemOnMap).toHaveBeenCalledWith({
        categories: mockedCategoriesState,
        pointingCategoryId: null,
        visibleItems: mockedVisibleItems,
        onAdd: expect.any(Function),
      });
    });
  });

  describe('when open for creating and new marker passed', () => {
    test('renders successfully with mocked modal', () => {
      // Arange
      const mockedUsePlay = jest.fn(() => ({
        ...mockedUsePlayResult,
        pointingCategoryId: mockedPointingCategoryId,
      }));
      (usePlay as unknown as jest.Mock).mockImplementation(mockedUsePlay);
      const mockedUseAddItemOnMap = jest.fn(() => ({
        ...mockedUseAddItemOnMapResult,
        allMarkers: [...mockedAllMarkers, mockedNewMarker],
        newMarker: mockedNewMarker,
      }));
      (useAddItemOnMap as unknown as jest.Mock).mockImplementation(
        mockedUseAddItemOnMap
      );
      // Act
      const { baseElement } = renderWithTheme(<PlayMap />);
      // Assert
      expect(baseElement).toMatchSnapshot();
      expect(mockedUsePlay).toHaveBeenCalledWith();
      expect(mockedUseAddItemOnMap).toHaveBeenCalledWith({
        categories: mockedCategoriesState,
        pointingCategoryId: mockedPointingCategoryId,
        visibleItems: mockedVisibleItems,
        onAdd: expect.any(Function),
      });
    });
  });

  describe('when clicks to edit item', () => {
    test('calls mockedOpenItemUpdating', async () => {
      // Arange
      const { getAllByLabelText } = renderWithTheme(<PlayMap />);
      // Act
      await userEvent.click(getAllByLabelText('edit-item')[0]);
      // Assert
      expect(mockedOpenItemUpdating).toHaveBeenCalledWith(mockedItem1.id);
    });
  });
});
