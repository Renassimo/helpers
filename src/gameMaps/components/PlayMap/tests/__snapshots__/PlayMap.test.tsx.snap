// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayMap renders successfully 1`] = `
<body>
  <div>
    <div>
      MockedMapOnImage 
      withOnClick
       
      /
       
      #fff
       - 
      <div>
        #fff
         - 
        0.5
         - 
        10.5
         - 
        existing
         
        - 
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Item 1 description
            </p>
          </div>
          <div
            class="MuiBox-root css-s2uf1z"
          >
            <button
              aria-label="edit-item"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div>
        #fff
         - 
        0.6
         - 
        10.6
         - 
        existing
         
        - 
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Item 2 description
            </p>
          </div>
          <div
            class="MuiBox-root css-s2uf1z"
          >
            <button
              aria-label="edit-item"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`PlayMap when open for creating and new marker passed renders successfully with mocked modal 1`] = `
<body>
  <div>
    <div>
      MockedMapOnImage 
      withOnClick
       
      /
       
      #fff
       - 
      <div>
        #fff
         - 
        0.5
         - 
        10.5
         - 
        existing
         
        - 
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Item 1 description
            </p>
          </div>
          <div
            class="MuiBox-root css-s2uf1z"
          >
            <button
              aria-label="edit-item"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div>
        #fff
         - 
        0.6
         - 
        10.6
         - 
        existing
         
        - 
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Item 2 description
            </p>
          </div>
          <div
            class="MuiBox-root css-s2uf1z"
          >
            <button
              aria-label="edit-item"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div>
        #888
         - 
        1
         - 
        2
         - 
        new
         
        - 
        new marker description
      </div>
    </div>
    <div>
      MockedItemFormModal 
      closed
       - 
      gm1
       -
       
      1
       - 
      2
       - 
      pl1
       - 
      cat2
    </div>
  </div>
</body>
`;

exports[`PlayMap when open for editing renders successfully with mocked modal 1`] = `
<body>
  <div>
    <div>
      MockedMapOnImage 
      withOnClick
       
      /
       
      #fff
       - 
      <div>
        #fff
         - 
        0.5
         - 
        10.5
         - 
        existing
         
        - 
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Item 1 description
            </p>
          </div>
          <div
            class="MuiBox-root css-s2uf1z"
          >
            <button
              aria-label="edit-item"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div>
        #fff
         - 
        0.6
         - 
        10.6
         - 
        existing
         
        - 
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Item 2 description
            </p>
          </div>
          <div
            class="MuiBox-root css-s2uf1z"
          >
            <button
              aria-label="edit-item"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      MockedItemFormModal 
      closed
       - 
      gm1
       -
       
      0.5
       - 
      10.5
       - 
      pl1
       - 
      cat1
    </div>
  </div>
</body>
`;
